SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -44,19 +44,62 @@ EntityFactory::EntityFactory(USpatialNetDriver* InNetDriver, USpatialPackageMapC
 {
 }
 
-TArray<FWorkerComponentData> EntityFactory::CreateEntityComponents(USpatialActorChannel* Channel, uint32& OutBytesWritten)
+FUnrealObjectRef GetStablyNamedObjectRef(UObject* Object)
+{
+	if (Object == nullptr)
+	{
+		return FUnrealObjectRef::NULL_OBJECT_REF;
+	}
+
+	// No path in SpatialOS should contain a PIE prefix.
+	FString TempPath = Object->GetFName().ToString();
+	TempPath = UWorld::RemovePIEPrefix(TempPath);
```

Comment:
If that is something extracted from NetworkRemapPath, I think the only part that could be relevant is the one related to StreamingLevel. We should check that this only relates to levels streamed at runtime, and not level composition in the editor.

Reply:
Hmm, I didn't notice the StreamingLevel case, but it's worth noting that we previously called NetworkRemapPath with `bIsReading = false` which skips a lot of the branches.

With the new code, NetworkRemapPath won't be called at runtime at all, it will be completely replaced with this RemovePIEPrefix thing. I did leave in some slow check code to verify it's doing the same thing, but slow checks are not very respected.

Reply:
I see. Probably fine then.